indexer/services/vulcan/__tests__/handlers/order-replace-handler.test.ts

```diff
@@ -0,0 +1,899 @@
+        APIOrderStatusEnum.BEST_EFFORT_OPENED,
+        expectSubaccountMessage,
+      );
+      expectStats();
+      // clear mocks
+      jest.clearAllMocks();
+
+      // Handle the order place off-chain update with the replacement order
+      await onMessage(orderReplacementMessage);
+
+      await checkOrderPlace(
+        expectedOrderUuid,
+        redisTestConstants.defaultSubaccountUuid,
+        expectedReplacedOrder,
+      );
+      expect(OrderbookLevelsCache.updatePriceLevel).not.toHaveBeenCalled();
+      if (hasCanceledOrderId) {
+        expect(CanceledOrdersCache.removeOrderFromCaches).toHaveBeenCalled();
+      }
+      await expectCanceledOrderStatus(expectedOrderUuid, CanceledOrderStatus.NOT_CANCELED);
+
+      expect(logger.error).not.toHaveBeenCalled();
+      expectWebsocketMessagesSent(
+        producerSendSpy,
+        expectedReplacedOrder,
+        dbOrder,
+        testConstants.defaultPerpetualMarket,
+        APIOrderStatusEnum.BEST_EFFORT_OPENED,
+        expectSubaccountMessage,
+      );
+      expectStats(true);
+    });
+
+    // TODO(IND-68): Remove this test once order replacement logic does not change price levels as
+    // orders are removed before being re-placed.
+    it.each([
+      [
+        'goodTilBlock',
+        redisTestConstants.defaultOrder,
+        replacementMessage,
+        redisTestConstants.defaultRedisOrder,
+        dbDefaultOrder,
+        redisTestConstants.defaultOrderUuid,
+        replacedOrder,
+        true,
+        true,
+      ],
+      [
+        'goodTilBlockTime',
+        redisTestConstants.defaultOrderGoodTilBlockTime,
+        replacementMessageGoodTilBlockTime,
+        redisTestConstants.defaultRedisOrderGoodTilBlockTime,
+        dbOrderGoodTilBlockTime,
+        redisTestConstants.defaultOrderUuidGoodTilBlockTime,
+        replacedOrderGoodTilBlockTime,
+        false,
+        true,
+      ],
+      [
+        'conditional',
+        redisTestConstants.defaultConditionalOrder,
+        replacementMessageConditional,
+        redisTestConstants.defaultRedisOrderConditional,
+        dbConditionalOrder,
+        redisTestConstants.defaultOrderUuidConditional,
+        replacedOrderConditional,
+        false,
+        true,
+      ],
+      [
+        'Fill-or-Kill',
+        redisTestConstants.defaultOrderFok,
+        replacementMessageFok,
+        redisTestConstants.defaultRedisOrderFok,
+        dbDefaultOrderFok,
+        redisTestConstants.defaultOrderUuid,
+        replacedOrderFok,
+        true,
+        false,
+      ],
+      [
+        'Immediate-or-Cancel',
+        redisTestConstants.defaultOrderIoc,
+        replacementMessageIoc,
+        redisTestConstants.defaultRedisOrderIoc,
+        dbDefaultOrderIoc,
+        redisTestConstants.defaultOrderUuid,
+        replacedOrderIoc,
+        true,
+        false,
+      ],
+    ])('handles order place for replacing order (with %s), resting on book', async (
```

Comment:
Is there a test for the orderbook message being sent for different priced replacement, but not sent for the same priced replacement?

Reply:
Added test!